src/assets/scripts/client/ui/VideoMapController.js

```diff
@@ -0,0 +1,220 @@
+         * @type {jquery|HTML Element}
+         * @default null
+         */
+        this.$dialog = null;
+
+        /**
+         * Currently selected maps, default to the one in the Airport file
+         *
+         * @property _selectedMaps
+         * @type {object}
+         * @default null
+         */
+        this._selectedMaps = null;
+
+
+        this.init()
+            ._setupHandlers()
+            .enable();
+    }
+
+    /**
+     *
+     * @for VideoMapController
+     * @method init
+     * @chainable
+     */
+    init() {
+        this.$dialog = $(UI_MAP_MODAL_TEMPLATE);
+        this.$dialogBody = this.$dialog.find(`.${CLASSNAMES.DIALOG_BODY}`);
+
+        this._buildDialogBody();
+        this.$element.append(this.$dialog);
+
+        return this;
+    }
+
+    /**
+     * Create event handlers
+     *
+     * Should be run once only on instantiation
+     *
+     * @for TrafficRateController
+     * @method _setupHandlers
+     * @chainable
+     */
+    _setupHandlers() {
+        this._onAirportChangeHandler = this.onAirportChange.bind(this);
+
+        return this;
+    }
+
+    /**
+     * Enable event handlers
+     *
+     * should be run only once on instantiation
+     *
+     * @for TrafficRateController
+     * @method enable
+     * @chainable
+     */
+    enable() {
+        this._eventBus.on(EVENT.AIRPORT_CHANGE, this._onAirportChangeHandler);
+
+        return this;
+    }
+
+    /**
+     * Disable event handlers
+     *
+     * @for TrafficRateController
+     * @method disable
+     * @chainable
+     */
+    disable() {
+        this._eventBus.off(EVENT.AIRPORT_CHANGE, this._onAirportChangeHandler);
+
+        return this;
+    }
+
+    /**
+     * Returns whether the airport selection dialog is open
+     *
+     * @for TrafficRateController
+     * @method isDialogOpen
+     * @return {boolean}
+     */
+    isDialogOpen() {
+        return this.$dialog.hasClass(SELECTORS.CLASSNAMES.OPEN);
+    }
+
+    /**
+    * @for TrafficRateController
+    * @method toggleDialog
+    */
+    toggleDialog() {
+        this.$dialog.toggleClass(SELECTORS.CLASSNAMES.OPEN);
+    }
+
+    /**
+     * Rebuilds the dialog body when the airport is changed.
+     *
+     * @for TrafficRateController
+     * @method onAirportChange
+     */
+    onAirportChange() {
+        this._buildDialogBody();
+    }
+
+    /**
+     * Builds the dialog body
+     *
+     * @for TrafficRateController
+     * @method _buildDialogBody
+     */
+    _buildDialogBody() {
+        this.$dialogBody.empty();
+        const airportModel = AirportController.airport_get();
+        const mapNames = airportModel.mapCollection.getMapNames();
+        this._selectedMaps = airportModel.mapCollection.getVisibleMapNames();
+
+        for (const mapName of mapNames) {
+            const isChecked = this._selectedMaps.includes(mapName);
+            const $formElement = this._buildRow(mapName, mapName, isChecked, this._onChangeSelectedMaps);
```

Comment:
if `_onChangeSelectedMaps` is already attached to the instance, do we need to pass it here?  secondly, this should be bound to the class if it's (eventually) going to be used as a hander (see comment below)